generic/lshw.py

```diff
@@ -0,0 +1,177 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM.
+# Author: Ramya BS <ramya@linux.vnet.ibm.com>
+
+import os
+import re
+import subprocess
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Lshwrun(Test):
+
+    """
+    lshw is a small tool to extract detailed information on the
+    hardware configuration of the machine.
+    It can report exact memory configuration,firmware version,
+    mainboard configuration, CPU version and speed,cache configuration, bus
+    speed, etc. on DMI-capable x86 or IA-64 systems and on some PowerPC
+    machines (PowerMac G4 is known to work).
+    """
+    interface = process.system_output("ip route show")
+    active_interface = re.search(
+        r"default via\s+\S+\s+dev\s+(\w+)\s+proto", interface).group(1)
+    is_fail = 0
+
+    def run_cmd(self, cmd):
+        cmd_result = process.run(cmd, ignore_status=True, sudo=True,
+                                 shell=True)
+        if cmd_result.exit_status != 0:
+            self.is_fail += 1
+        return
+
+    def setUp(self):
+        if os.geteuid() != 0:
+            self.error("This program requires super user priv to run.")
+        soft = SoftwareManager()
+        for package in ("lshw", "net-tools", "iproute", "pciutils"):
+            if not soft.check_installed(package) and not soft.install(package):
+                self.error("Fail to install %s required for this"
+                           " test." % package)
+
+    def test_lshw(self):
+        """
+        lshw without any options would generate full information
+        report about all detected hardware.
+        """
+        self.log.info("===============Executing lshw test ===="
+                      "===========")
+        self.run_cmd("lshw")
+        self.run_cmd("lshw -version")
+        if self.is_fail >= 1:
+            self.fail("%s command(s) failed to execute  "
+                      "verification" % self.is_fail)
+
+    def test_lshw_short(self):
+        """
+        With "-short" option,the lshw command would generate a brief
+        information report about the hardware devices
+        """
+        self.log.info("===============Executing lshw -short tests ===="
+                      "===========")
+        self.run_cmd("lshw -short")
+        self.run_cmd("lshw -short  -class network")
+        self.run_cmd("lshw -short  -class storage")
+        self.run_cmd("lshw -short  -class memory")
+        self.run_cmd("lshw -short  -class power")
+        self.run_cmd("lshw -short  -class bus")
+        self.run_cmd("lshw -short  -class processor")
+        self.run_cmd("lshw -short  -class system")
+        if self.is_fail >= 1:
+            self.fail("%s command(s) failed to execute "
+                      "verification" % self.is_fail)
```

Comment:
it will good if we show user which command is fail for that `self.fail() `can be use in` run_cmd` helper function what you think ?